Artifacts/windows-install-certificate/certificate-install.ps1

```diff
@@ -0,0 +1,109 @@
+Param(
+    [ValidateNotNullOrEmpty()]
+    [string] $certificateName,
+    [ValidateNotNullOrEmpty()]
+    [string] $base64cert,
+    [ValidateNotNullOrEmpty()]
+    [string] $certificatePassword
+)
+
+
+##################################################################################################
+
+#
+# Powershell Configurations
+#
+
+# Note: Because the $ErrorActionPreference is "Stop", this script will stop on first failure.  
+#       This is necessary to ensure we capture errors inside the try-catch-finally block.
+$ErrorActionPreference = "Stop"
+
+# Ensure we set the working directory to that of the script.
+pushd $PSScriptRoot
+
+# Ensure that current process can run scripts. 
+Set-ExecutionPolicy -ExecutionPolicy Bypass -Scope Process -Force 
+
+#
+# Functions used in this script.
+#
+
+function Handle-LastError
+{
+    [CmdletBinding()]
+    param(
+    )
+
+    $message = $error[0].Exception.Message
+    if ($message)
+    {
+        Write-Host -Object "ERROR: $message" -ForegroundColor Red
+    }
+    
+    # IMPORTANT NOTE: Throwing a terminating error (using $ErrorActionPreference = "Stop") still
+    # returns exit code zero from the PowerShell script when using -File. The workaround is to
+    # NOT use -File when calling this script and leverage the try-catch-finally block and return
+    # a non-zero exit code from the catch block.
+    exit -1
+}
+
+#
+# Handle all errors in this script.
+#
+
+trap
+{
+    # NOTE: This trap will handle all errors. There should be no need to use a catch below in this
+    #       script, unless you want to ignore a specific error.
+    Handle-LastError
+}
+
+###################################################################################################
+
+#
+# Main execution block.
+#
+
+try
+{
+    Write-Host "Installing certificate $certificateName"
+
+    If(-NOT ([Security.Principal.WindowsPrincipal] [Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole] "Administrator"))
+    {
+        #build up the deploy arguments
+        $arguments = "-file `"{0}`"" -f $script:MyInvocation.MyCommand.Path
+    
+        # Start the new process
+        Start-Process powershell.exe -Verb runas -ArgumentList $arguments
+        exit
+    }
+    else
+    {
+        $securePassword = ConvertTo-SecureString -String $certificatePassword -AsPlainText -Force
+		$certificatePassword = "deleted"
+
+        $tempFilePath = [System.IO.Path]::GetTempFileName()
+        Write-Host "Temp file path '$tempFilePath'" 
+
+        [System.IO.File]::WriteAllBytes($tempFilePath, [System.Convert]::FromBase64String($base64cert))
+        Write-Host "Certificate saved"
+        		
+        Get-ChildItem -Path $tempFilePath | Import-PfxCertificate -CertStoreLocation Cert:\LocalMachine\My -Exportable -Password $securePassword
+        Write-Host "Certificate $certificateName added to the LocalMachine\My store succesfully"
+
+        [System.IO.File]::Delete($tempFilePath)
```

Comment:
Consider using `Remove-Item -Path "$tempFilePath" -Force`, as it will not throw.

Reply:
Implemented in the latest commit.